nav2_costmap_2d/src/costmap_2d_cloud.cpp

```diff
@@ -154,7 +154,7 @@ void voxelCallback(const nav2_msgs::msg::VoxelGrid::ConstSharedPtr grid)
       // uint32_t a = g_colors_a[c.status] * 255.0;
 
       uint32_t col = (r << 16) | (g << 8) | b;
-      cval = *reinterpret_cast<float *>(&col);
+      memcpy(&cval, &col, sizeof col);
```

Comment:
I don't think this is valid. Most likely cval never had memory allocated to copy into. You are also changing behavior here; adding work that probably isn't necessary.

What was the compiler warning?

Reply:
Never mind. I went through the code to understand the context, and I think your approach is good.

Reply:
It was warning about strict aliasing. I found this method of handling it.